### ISSUE TITLE
Cannot rebuild sage

### ISSUE THREAD

**Issue by mkoeppe** [3-manifolds/sagedocker#2]
The Sage installation in computop/sage:9.5 has several issues that prevent rebuilding when changes are made to Sage. 

I ran into this while setting up computop/sage as a VS Code devcontainer for Sage development, https://trac.sagemath.org/ticket/33671

The issues and workarounds for them are described in:
https://git.sagemath.org/sage.git/diff/.devcontainer/devcontainer-computop-sage.json?id=8ec5393e16da7b211d4ee060b1dc34dc0facae52

**Comment by NathanDunfield** [comment]
Thanks for reporting this Matthias, but could you expand on what is wrong?  I looked at the json file you linked to, but I didn't know that the `computop/sage` image supported rebuilding Sage at all and `pari` seems to work just fine for me.  Our [script](https://github.com/3-manifolds/sagedocker/blob/master/sage/scripts/01_build) for building Sage just goes with the defaults, so I'm not sure why rebuilding `pari` and `nauty` would fix anything.

**Comment by mkoeppe** [comment]
I think after installing Sage, some of the following scripts in https://github.com/3-manifolds/sagedocker/tree/master/sage/scripts clobber parts of the Sage installation.

**Comment by NathanDunfield** [comment]
Hmm, we do add some additional software to the `$SAGE_LOCAL` tree, mostly various topology Python packages, and also mess with the Jupyter(-Lab) config.   So that I can debug, what's a specific command that results in an error?

**Comment by mkoeppe** [comment]
> I didn't know that the `computop/sage` image supported rebuilding Sage at all

True, what is in `/sage` is not sufficient to do any rebuilds. `/sage/src` has the complete sources but the top-level Makefile and configure script are missing and `sage -b` just fails with an unhelpful error.

The devcontainer script uses its own source tree mounted in the container and configures it to re-use the /sage/local installation prefix.

**Comment by culler** [comment]
I am not sure if this is relevant, but there is now a symlink
sage/local/venv to a venv which contains a lot of the stuff that used to go
in sage/local.  I am guessing that whatever we put into sage/local now
needs to move to the venv.

- Marc
